4_basic_assembly/2_bitwise_operations/2_practical_bit_games/answers/3_write_code/6_2.asm

```diff
@@ -0,0 +1,45 @@
+; Basic Assembly
+; ==============
+;
+; 6.  Bonus: Identifying powers of two.
+;   
+;     6.2   Try to write that program again, this time without any loops.
+;           
+;           HINT: try to decrease the original number by 1.
+; 
+; Reference: http://aggregate.ee.engr.uky.edu/MAGIC/#Is%20Power%20of%202
+; x is a power of 2 iff (x AND (x-1)) == 0

```

Comment:
An extra check is required: Make sure that x != 0, as 0 is not a power of 2, but it will pass the code below.